src/app/components/selection/selection.component.ts

```diff
@@ -15,12 +16,12 @@ export class SelectionComponent {
   constructor(private weatherData: WeatherDataService) {}
 
   submit() {
-    this.weatherData.load(this.city).subscribe((data) => {
-      this.weather.city = data['name'];
-      this.weather.conditions = data['weather'][0]['main'];
-      this.weather.temperature = Math.round(data['main']['temp']);
+    this.weatherData.load(this.city).subscribe((data: WeatherLoadResponse) => {
```

Comment:
response we receive from weather API is better to have with strict type, that is why I have created WeatherLoadResponse